csharp/src/Drivers/Apache/Spark/SparkConnection.cs

```diff
@@ -719,28 +719,31 @@ private static void SetPrecisionScaleAndTypeName(short colType, string typeName,
                 case (short)ColumnTypeId.NCHAR:
                     {
                         bool success = new SqlCharTypeParser().TryParse(typeName, out SqlCharVarcharParserResult? result);
-                        tableInfo?.Precision.Add(success ? result!.ColumnSize : SqlVarcharTypeParser.VarcharColumnSizeDefault);
+                        tableInfo?.Precision.Add(success && result != null ? result.ColumnSize : SqlVarcharTypeParser.VarcharColumnSizeDefault);
                         tableInfo?.Scale.Add(null);
-                        tableInfo?.BaseTypeName.Add(success ? result!.BaseTypeName : "CHAR");
+                        tableInfo?.BaseTypeName.Add(success && result != null ? result.BaseTypeName : "CHAR");
                         break;
                     }
+
                 case (short)ColumnTypeId.VARCHAR:
                 case (short)ColumnTypeId.LONGVARCHAR:
                 case (short)ColumnTypeId.LONGNVARCHAR:
                 case (short)ColumnTypeId.NVARCHAR:
                     {
                         bool success = new SqlVarcharTypeParser().TryParse(typeName, out SqlCharVarcharParserResult? result);
-                        tableInfo?.Precision.Add(success ? result!.ColumnSize : SqlVarcharTypeParser.VarcharColumnSizeDefault);
+                        tableInfo?.Precision.Add(success && result != null ? result.ColumnSize : SqlVarcharTypeParser.VarcharColumnSizeDefault);
```

Comment:
result seems to be nullable type, we could use null-coalescing operator to make this concise
```suggestion
                        tableInfo?.Precision.Add(success && result?.ColumnSize ?? SqlVarcharTypeParser.VarcharColumnSizeDefault);
```

Reply:
Fixed by calling the Parse method, instead. Will throw if unable to parse the type name string.